src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/TraceReporter.kt

```diff
@@ -18,23 +18,44 @@ import kotlin.math.*
 internal fun StringBuilder.appendTrace(
     scenario: ExecutionScenario,
     results: ExecutionResult?,
-    trace: Trace
+    trace: Trace,
+    exceptionStackTraces: Map<Throwable, ExceptionNumberAndStacktrace>
 ) {
-    val startTraceGraphNode = constructTraceGraph(scenario, results, trace)
+    val startTraceGraphNode = constructTraceGraph(scenario, results, trace, exceptionStackTraces)
 
     appendln(PARALLEL_PART_TITLE)
     val traceRepresentation = traceGraphToRepresentationList(startTraceGraphNode, false)
     appendTraceRepresentation(scenario, traceRepresentation)
     appendln()
-
     appendln()
+
+    if (exceptionStackTraces.isNotEmpty()) {
```

Comment:
You append exception stack traces for both stress testing and model checking modes. This information is not related to the interleaving traces. I would, therefore, split this function to "appendShortTrace" and "appendDetailedTrace" functions, calling them separately. This way, you can easily print the stack traces between the traces.